Check if QNetworkInformation::loadBackendByFeatures is successful otherwise
avoid nullptr access

Index: src/pimcommon/network/networkmanager.cpp
--- src/pimcommon/network/networkmanager.cpp.orig
+++ src/pimcommon/network/networkmanager.cpp
@@ -14,9 +14,11 @@ Q_GLOBAL_STATIC(NetworkManager, s_pNetworkManagerSelf)
 NetworkManager::NetworkManager(QObject *parent)
     : QObject(parent)
 {
-    QNetworkInformation::instance()->loadBackendByFeatures(QNetworkInformation::Feature::Reachability);
-    connect(QNetworkInformation::instance(), &QNetworkInformation::reachabilityChanged, this, &NetworkManager::refreshStatus);
-    connect(QNetworkInformation::instance(), &QNetworkInformation::isBehindCaptivePortalChanged, this, &NetworkManager::refreshStatus);
+    bool isBackendLoaded = QNetworkInformation::loadBackendByFeatures(QNetworkInformation::Feature::Reachability);
+    if (isBackendLoaded) {
+        connect(QNetworkInformation::instance(), &QNetworkInformation::reachabilityChanged, this, &NetworkManager::refreshStatus);
+        connect(QNetworkInformation::instance(), &QNetworkInformation::isBehindCaptivePortalChanged, this, &NetworkManager::refreshStatus);
+    }
     refreshStatus();
 }
 
